tests/acceptance/cli_test.py

```diff
@@ -718,6 +718,15 @@ def test_up_handles_force_shutdown(self):
         os.kill(proc.pid, signal.SIGTERM)
         wait_on_condition(ContainerCountCondition(self.project, 0))
 
+    @v2_only()
+    def test_up_handles_abort_on_container_exit(self):
+        self.base_dir = 'tests/fixtures/sleeps-composefile'
+        start_process(
+            self.base_dir,
+            ['-f', 'one-exits.yml', 'up', '-t', '200', '--abort-on-container-exit'])
+        wait_on_condition(ContainerCountCondition(self.project, 2))
+        wait_on_condition(ContainerCountCondition(self.project, 0), timeout=1)
```

Comment:
Ah, I guess this is why it needs to `sleep 0.5`, although on a loaded jenkins instance, this could still fail, even with the sleep. The `timeout=1` is also likely to flake a lot.

I'd like to propose this testing strategy using the `top` command.

```yaml
# Set project so teardown does the correct cleanup, you can find examples of this in other tests
self._project = get_project(self.base_dir, ['one-exits.yml'])
start_process(...)

wait_on_condition(ContainerCountCondition(self.project, 2))
self.project.stop(['simple'])
wait_on_condition(ContainerCountCondition(self.project, 0))
```